keps/sig-storage/3751-volume-provisioned-io/README.md

```diff
@@ -0,0 +1,640 @@
+
+###### Will enabling / using this feature result in any new API calls?
+
+Yes.
+
+- API call type: PATCH PVC
+- estimated throughput: low, only once for PVCs that have
+  iops/throughput field gets updated in PVC
+- originating component(s): kube-controller-manager
+
+
+###### Will enabling / using this feature result in introducing new API types?
+
+Yes. ControllerModifyVolume.
+
+
+###### Will enabling / using this feature result in any new calls to the cloud provider?
+
+Yes. ControllerModifyVolume.
+
+###### Will enabling / using this feature result in increasing size or count of the existing API objects?
+
+
+###### Will enabling / using this feature result in increasing time taken by any operations covered by existing SLIs/SLOs?
+
+
+###### Will enabling / using this feature result in non-negligible increase of resource usage (CPU, RAM, disk, IO, ...) in any components?
+
+
+### Troubleshooting
+
+<!--
+This section must be completed when targeting beta to a release.
+
+For GA, this section is required: approvers should be able to confirm the
+previous answers based on experience in the field.
+
+The Troubleshooting section currently serves the `Playbook` role. We may consider
+splitting it into a dedicated `Playbook` document (potentially with some monitoring
+details). For now, we leave it here.
+-->
+
+###### How does this feature react if the API server and/or etcd is unavailable?
+
+
+###### What are other known failure modes?
+
+
+<!--
+For each of them, fill in the following information by copying the below template:
+  - [Failure mode brief description]
+    - Detection: How can it be detected via metrics? Stated another way:
+      how can an operator troubleshoot without logging into a master or worker node?
+    - Mitigations: What can be done to stop the bleeding, especially for already
+      running user workloads?
+    - Diagnostics: What are the useful log messages and their required logging
+      levels that could help debug the issue?
+      Not required until feature graduated to beta.
+    - Testing: Are there any tests for failure mode? If not, describe why.
+-->
+
+###### What steps should be taken if SLOs are not being met to determine the problem?
+
+
+## Implementation History
+
+<!--
+Major milestones in the lifecycle of a KEP should be tracked in this section.
+Major milestones might include:
+- the `Summary` and `Motivation` sections being merged, signaling SIG acceptance
+- the `Proposal` section being merged, signaling agreement on a proposed design
+- the date implementation started
+- the first Kubernetes release where an initial version of the KEP was available
+- the version of Kubernetes where the KEP graduated to general availability
+- when the KEP was retired or superseded
+-->
+
+## Drawbacks
+
+None that we are aware of
+
+## Alternatives
+
+* Modify the storage class in the PVC. See this [document](https://docs.google.com/document/d/1d5Ch54t8-uyORh56AP7rN9AkQfKuLBsCrrBjOohbWLA/edit?usp=sharing).
```

Comment:
It would be good to add more details on why these alternative approaches were not chosen.

Reply:
This is actually the using a map proposal and we are discussing it right now.